pandas/io/html.py

```diff
@@ -920,6 +921,7 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
     return ret
 
 
+@deprecate_nonkeyword_arguments(version="1.0", allowed_args=["io", "match"])
```

Comment:
Why are we allowing `match` as a positional argument?

I'd like not to repeat the arguments here. That seems fragile. Is it possible to have the decorator inspect the signature and warn on any positional_or_keyword arg passed positionally?